<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0105EN-SkillsNetwork/labs/Module2/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>


#### Add your code below following the instructions given in the course


# My Jupyter Notebook on IBM Watson Studio

**David Hegarty**

I am currently working as a Finance Operations Systems Specialist

*I am interested in data science because I want to be better equipped to analyze and improve business processes. I want to find the simplest user experience that also generates the best and most powerful reporting data in the back-end that can accurately highlight and describe where processes may have issues that can be targeted to improve the processes.*

### The code below calculates the fibonacci number of a user given number

In [44]:
def nth_fibonacci(Number):
    """
    Return the nth fibonacci number based on user input.
    """
    if n < 2:
        return n
    fs = [0, 1]
    for i in range(1, n):
        fs.append(fs[i] + fs[i - 1])
    return fs[n]


# get user input
n = input()

# declaring a flag 'validInt' for check
valid_int = True
try:
    # try to convert the input to an integer
    n = int(n)
except ValueError:
    # raise a value error and set valid_int to false if the input cannot be converted
    valid_int = False

# check if the input was converted and print the relevant output
if valid_int:
    # check if the integer provided is negative and print an appropriate result
    if int(n) < 0:
        print("This code is not designed to calculate the negative values of the fibonacci series")
    else:
        print("Fibonacci number", n, "= ", nth_fibonacci(n))
else:
    print("Sorry I cannot calulate the fibonacci number from the given input: "
          , n, "\nPlease enter an integer to be calculated next time.")

 15


Fibonacci number 15 =  610


---
## Code Cell Usage

- When you run the codecell above, it will prompt you for an input
- While this input can technically be anything, it will only calculate the fibonacci number or an integer input.
- Eg. '1', not 'one'
- The try catch block below shows the use of error handling to validate user input:

```python
# declaring a flag 'validInt' for check
valid_int = True
try:
    # try to convert the input to an integer
    n = int(n)
except ValueError:
    # raise a value error and set valid_int to false if the input cannot be converted
    valid_int = False
```

- The table below details the first 20 fibonacci numbers for testing validation purposes:

| Input Integer | Expected fibonacci number in output |
| ----------- | ----------- |
| 1 | 0 |
| 2 | 1 |
| 3 | 1 |
| 4 | 2 |
| 5 | 3 |
| 6 | 5 |
| 7 | 8 |
| 8 | 13 |
| 9 | 21 |
| 10 | 34 |
| 11 | 55 |
| 12 | 89 |
| 13 | 144 |
| 14 | 233 |
| 15 | 377 |
| 16 | 610 |
| 17 | 987 |
| 18 | 1597 |
| 19 | 2584 |
| 20 | 4181 |